crates/flycheck/src/lib.rs

```diff
@@ -297,8 +297,12 @@ impl FlycheckActor {
                 let mut cmd = Command::new(toolchain::cargo());
                 cmd.arg(command);
                 cmd.current_dir(&self.root);
-                cmd.args(["--workspace", "--message-format=json", "--manifest-path"])
-                    .arg(self.root.join("Cargo.toml").as_os_str());
+                cmd.args([
+                    "--workspace",
+                    "--message-format=json-diagnostic-rendered-ansi",
```

Comment:
This should be gated behind a config disabled by default, otherwise this may break clients that don't handle color codes.

Reply:
good point, I had forgotten about how this would impact other clients. I guess it should be called something like `checkOnSave_enableColorOutput` ?

Reply:
Hmm, actually, thinking about again, a config might make sense for the client, but not the server. We might want to make this an experimental client capability instead.

Reply:
I've just pushed a change as suggested. Should this *also* be exposed as a config in `package.json`? Do I need to update #4604 and/or `lsp-extensions.md` to document this as a VSCode-specific extension?

Reply:
Would be good to add that to the markdown file as a small entry yes, most likely VSCode is the only client benefiting from this, but its always good to keep that list in sync.